src/SimpleGraphs/generators/randgraphs.jl

```diff
@@ -989,6 +989,144 @@ function uniform_tree(n::Integer; rng::Union{Nothing,AbstractRNG}=nothing)
     return prufer_decode(random_code)
 end
 
+"""
+    havel_hakimi_graph_generator(degree_sequence::AbstractVector{<:Integer})
+
+Returns a simple graph with a given finite degree sequence of non-negative integers generated via the Havel-Hakimi algorithm which works as follows:
+1. successively connect the node of highest degree to other nodes of highest degree;
+2. sort the remaining nodes by degree in decreasing order;
+3. repeat the procedure.
+
+## References
+1. [Hakimi (1962)](https://doi.org/10.1137/0110037);
+2. [Wikipedia](https://en.wikipedia.org/wiki/Havel%E2%80%93Hakimi_algorithm).
+"""
+function havel_hakimi_graph_generator(degree_sequence::AbstractVector{<:Integer})
```

Comment:
```suggestion
function havel_hakimi_graph(degree_sequence::AbstractVector{<:Integer})
```
would maybe be shorter?

Reply:
It would be nice, if we could specify the `eltype` of the resulting graph instead of only allowing `Int`. I would write two methods for this function:

```julia
function havel_hakimi_graph_generator(degree_sequence::AbstractVector{<:Integer})
    return havel_hakimi_graph_generator(Int, degree_sequence)
end

function havel_hakimi_graph_generator(T::Type{<:Integer}, degree_sequence::AbstractVector{<:Integer})
    [...]
    graph = SimpleGraph{T}(length(degree_sequence))
    [...]
end

then one could optionally specify the eltype.

Reply:
The `eltype` behavior you suggested should be implemented now.